src/explicitode.jl

```diff
@@ -208,24 +208,45 @@ end
 
 # taylorinteg
 doc"""
-    taylorinteg(f, x0, t0, tmax, order, abs_tol; keyword... )
+    taylorinteg(f, x0, t0, tmax, order, abstol; keyword... )
 
 This is a general-purpose Taylor integrator for the explicit ODE
 $\dot{x}=f(t,x)$ with initial condition specified by `x0` at time `t0`.
 It returns a vector with the values of time (independent variable),
 and a vector (of type `typeof(x0)`) with the computed values of
-the dependent variables. The integration stops when time
-is larger than `tmax`, or the number of saved steps is larger
+the dependent variable(s). The integration stops when time
+is larger than `tmax` (in which case the last returned values are
+`t_max`, `x(t_max)`), or else when the number of saved steps is larger
 than `maxsteps`.
 
 The integrator uses polynomial expansions on the independent variable
-of order `order` and the parameter `abs_tol` serves to define the
+of order `order`; the parameter `abstol` serves to define the
 time step using the last two Taylor coefficients of the expansions.
 
 The current keyword argument is `maxsteps=500`.
+
+Example:
+
+`using TaylorIntegration`
+
+`f(t, x) = x^2`
+
+`tv, xv = taylorinteg(f, 3, 0.0, 0.3, 25, 1.0e-20, maxsteps=100 )`
+
+---
 """
-function taylorinteg{T<:Number}(f, x0::T, t0::T, t_max::T,
-        order::Int, abs_tol::T; maxsteps::Int=500)
+function taylorinteg{S<:Number, T<:Number, U<:Number, V<:Number}(f, x0::S,
```

Comment:
Well, perhaps in the future we could add support for exotic users who prefer working with `Rational{BigInt}`'s instead of `Float64`'s! But for the time being I agree that all variables should be at least `Float64`'s. So I've changed the promotions in the code for the following:

```julia
# in the line below, x0::T with T<:Number
x0, t0, tmax, abstol, afloat = promote(x0, t0, tmax, abstol, one(Float64))
# and something analogous when x0::Array{T,1}, with T<:Number
```

Is this what you were suggesting?